tests/main/test_grid_init.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import pytest
+
+import pace.util
+from pace.util.grid import MetricTerms
+
+
+def get_cube_comm(layout, rank: int):
+    return pace.util.CubedSphereCommunicator(
+        comm=pace.util.NullComm(rank=rank, total_ranks=6 * layout[0] * layout[1]),
+        partitioner=pace.util.CubedSpherePartitioner(
+            pace.util.TilePartitioner(layout=layout)
+        ),
+    )
+
+
+def get_quantity_factory(layout, nx_tile, ny_tile, nz):
+    nx = nx_tile // layout[0]
+    ny = ny_tile // layout[1]
+    return pace.util.QuantityFactory(
+        sizer=pace.util.SubtileGridSizer(
+            nx=nx, ny=ny, nz=nz, n_halo=3, extra_dim_lengths={}
+        ),
+        numpy=np,
+    )
+
+
+@pytest.mark.parametrize("rank", [0, 1, 4])
```

Comment:
Do we not need to test all 9?

Reply:
The MPI-aware test will test all 54 ranks, this is only added here as a fast early warning in case something is horribly wrong. These three ranks should test all "kinds" of ranks (corner, edge, and center).